### PR TITLE
Expand RQ docs: Deal with failed jobs and dirty counts

### PR DESCRIPTION
**Pull request by unho**

### CODE REVIEW

**Comment by dwaynebailey** [comment]
lgtm

These docs seem to confirm that we actually do need a command to cleanly clear these and reset a TP for recounting.

**Comment by unho** [comment]
Yes, there is some discussion about that in #4219.

**Comment by unho** [comment]
> These docs seem to confirm that we actually do need a command to cleanly clear these and reset a TP for recounting.

Resetting a TP for recounting sounds easy. The problem is when we have to deal with dirty counts in upper levels (language, project) since the dirty counts might be right (something is being refreshed on other TP, or it is just zero), or might be wrong because this or other TP stats calculation failed leaving wrong dirty counts. I think is already being discussed on #4219.

File: docs/server/rq.rst [hunk 1]
```diff
@@ -77,6 +77,46 @@ To push failed jobs back into the queue we simply run the
 :djadmin:`retry_failed_jobs` management command.
 
 
+Delete all failed jobs
+++++++++++++++++++++++
+
+Sometimes some failed jobs no longer apply since they refer to removed items,
+so no matter how many times you run them they will keep failing. Note that
+sometimes those unrecoverable failed jobs are in company of other failed jobs
+that can be just re-run them by using the :djadmin:`retry_failed_jobs`
```
Comment (julen) [inline docs/server/rq.rst]: *that can be re-run by using...*

File: docs/server/rq.rst [hunk 2]
```diff
@@ -103,3 +143,32 @@ that all else is good and that the stats are fine or to be generated again.
...
 
    $ redis-cli -n 2 del pootle:refresh:stats
+
+
+Delete dirty counts
++++++++++++++++++++
+
+Sometimes statistics are correctly calculated, but the banner telling that
+stats are being refreshed doesn't dissappear. This usually happens because some
+job failed and didn't decrease the dirty counts.
```
Comment (julen) [inline docs/server/rq.rst]: From my experience this is due to an error in human intervention, e.g. workers were killed while there were ongoing jobs.

File: docs/server/rq.rst [hunk 3]
```diff
@@ -103,3 +143,32 @@ that all else is good and that the stats are fine or to be generated again.
...
+
+Make sure that there are no pending jobs or jobs being run since those could
+have increased the dirty counts. Re-run failed jobs if any.
+
+In order to delete all the dirty counts you must **stop the workers**.
+
+Remove the lastjob info for all dirty items:
```
Comment (julen) [inline docs/server/rq.rst]: \`\`lastjob\`\`

**Comment by julen** [comment]
Looks good to me.

File: docs/server/rq.rst (hunk 2 above)
Comment (unho) [inline docs/server/rq.rst]: Rephrasing as "some job failed to complete".

**Comment by unho** [comment]
Addressed raised concerns. Will land after I get @ta2-1 approval.

File: docs/server/rq.rst [hunk 4]
```diff
@@ -77,6 +77,46 @@ To push failed jobs back into the queue we simply run the
 :djadmin:`retry_failed_jobs` management command.
 
 
+Delete all failed jobs
+++++++++++++++++++++++
+
+Sometimes some failed jobs no longer apply since they refer to removed items,
+so no matter how many times you run them they will keep failing. Note that
```
Comment (phlax) [inline docs/server/rq.rst]: nit: replace "Sometimes some" with "Sometimes"

**Comment by phlax** [comment]
lgtm - although i cant vouch for any of the redis commands

File: docs/server/rq.rst [hunk 5]
```diff
@@ -77,6 +77,45 @@ To push failed jobs back into the queue we simply run the
...
+re-run by using the :djadmin:`retry_failed_jobs` management command:
+
+.. code-block:: bash
+
+   $ pootle retry_failed_jobs
+
+
+In order to delete all the failed jobs you must first **stop the workers**.
+
+Once the workers are stopped make sure that there are no failed job that you
```
Comment (ta2-1) [inline docs/server/rq.rst]: @unho s/job/jobs/

**Comment by ta2-1** [comment]
@unho, that's what we have discussed. good summary and gtm.

**Comment by unho** [comment]
Thanks everybody for the reviews. Merged.

**Pull request merged**